jsonschema-core.xml

```diff
@@ -455,6 +455,68 @@
 
         </section>
 
+        <section title="Principles">
+            <t>
+                JSON Schema is built on several principles, which both explain various design
+                choices, and provide guidance on how to extend the specification effectively.
+            </t>
+            <!-- Links will be added to relevant spec parts after the sections are reorganized
+                 to create the new schema loading section -->
+            <t>
+                <list>
+                    <t>
+                        All syntactically legal JSON documents with well-defined behavior are fully
+                        supported as instance data.  Notably, JSON objects with duplicate property
+                        names are syntactically legal but not well-defined.
+                    </t>
+                    <t>
+                        JSON Schema is defined over a data model, not JSON text.  Therefore
+                        JSON Schema can be used with any data format that can be reliably parsed
+                        into the data model.
+                    </t>
+                    <t>
+                        Locations in schemas and instances are always identified by URIs,
+                        JSON Pointers, or Relative JSON Pointers.  JSON Schema treats URIs as
+                        identifiers only.  Locating and retrieving URI-identified resources
+                        is outside of the scope of this specification.
+                    </t>
+                    <t>
+                        Schema URIs referenced or defined in JSON Schema's Core vocabulary MUST be
+                        resolvable without instance data.
```

Comment:
What does "resolvable without instance data" mean?

Reply:
Pretty much what it literally says- don't overthink it ;-)

A URI reference in `$ref` or `$id` is resolved against the base URI, which is the nearest parent `$id`.  If there isn't one, the schema resource's retrieval URI is used (per RFC 3986 generic rules for establishing base URIs).

What this point was trying to establish is that you can safely pre-process a schema so that all of those URI references are converted to full URIs (starting with a scheme) before you're given any instance data.  This makes applying the schema to any instance faster and simpler, because you don't have to keep track of base URIs and resolve references when you encounter them.

--------

However, this point doesn't quite work in 2019-09: the URI in `$recursiveRef` can't, based on the implementation mechanism described, be resolved like this, because what `$recursiveAnchor` does is replace the original base URI (from `$id` or the retrieval URI) with the base URI of the dynamic-scope outermost schema object containing `"$recursiveAnchor": true`, if any such schema object exists in the dynamic scope.

We could actually change the internal mechanism for this to be simply "replace the entire resolved reference from `$recursiveRef` with the URI of the dynamic-scope outermost etc.", because right now the only legal value of `$recursiveRef` is `"#"`, and both of these mechanisms produce the same result with that URI reference.

If we were to lift that restriction and allow any URI reference as a `$recursiveRef` value, then the two mechanisms produce different results, and require keeping track of base URIs at runtime.

See #868 for more details and hopefully eventually a decision on whether to require static (without instance) resolution for all keywords and a change to `$recursive*`, or whether to change this architectural principle.